packages/lodestar-validator/src/api/impl/rest/beacon/beacon.ts

```diff
@@ -39,12 +41,13 @@ export class RestBeaconApi implements IBeaconApi {
     return this.config.types.ForkResponse.fromJson(await this.client.get<Json>("/fork"), {case: "snake"});
   }
 
-  public async getGenesisTime(): Promise<Number64> {
+  public async getGenesis(): Promise<Genesis|null> {
     try {
-      return await this.client.get<Number64>("/genesis_time");
+      const genesisResponse = await this.clientV2.get<{data: Json}>("/genesis");
+      return this.config.types.Genesis.fromJson(genesisResponse.data, {case: "snake"});
     } catch (e) {
       this.logger.error("Failed to obtain genesis time", e);
-      return 0;
+      return null;
```

Comment:
Why return null instead of throwing the error? The spec allows returning 404 or 500 
https://github.com/ethereum/eth2.0-APIs/blob/1c6b9cb4ec8d8587f5739dd2d491a8742be89e48/apis/beacon/genesis.yaml#L25

Reply:
I don't wanna throw rest errors here since we are using this directly in validator so we would have to catch and differentiate errors which leads to ugly code. 

We might also expose this over graphql or rpc where rest errors wouldn't make sense.

Reply:
Lion makes a good point. We will need to handle the error case appropriately, eventually. Right now, `pollBeaconNode` silently does _nothing_ if this errors.
We can tackle this in another PR, seems like that has been the behavior even before this PR :grimacing: